### ISSUE TITLE
Decide whether to switch to big endian for serialization of long tag values.

### ISSUE THREAD

**Issue by sebright** [census-instrumentation/opencensus-java#519]
See the discussion in #511 (starting at https://github.com/census-instrumentation/opencensus-java/pull/511#discussion_r132809661).

**Comment by adriancole** [comment]
TL;DR; I asked about this playing devil's advocate.. mainly to nag us to
cite rationale whatever that may be.

The main idea here is that since this is a java repo, there's a bias
towards things done in java (not the general world of serialization). There
are lots of good reasons to not follow these conventions. Documenting
rationale that contrasts with
java norms, is helpful, as it makes things that might be obvious to us,
obvious to everyone.

I don't think it is a great idea, but there are a lot of encodings where
each side is assumed to be the same language
(ex finagle encodes some data literally with java serialization! or at
least it used to). Thrift defaults to big endian, and this
resulted in a bunch of questions from non-java users (but no questions from
java users at least to me anyway)

One way out is to place a disclaimer in the file that this stands as a
substitute for a not-yet-created polyglot spec for
the encodings, and that decisions around this prefer X which may not match
java norms. It almost doesn't matter what
the rationale is as long as it is mentioned somewhere.

To finish, I understand I'm making a mountain out of a molehill, but mainly
because I want to remind folks not everyone
knows what's the larger scope of this work. We are doing a good job for
sure, just sometimes we skip steps that are
easy enough to fix. Sometimes I ask as devil's advocate in attempts to help
remind us about things like this.

**Comment by dinooliva** [comment]
Adrian - That's a reasonable point, we can note in the comments that this is our language-independent encoding of TagContext that will also be used (minimally) by our C++ and Go implementations.

Unless there's a really big issue, changing the encoding is unlikely to happen at this point.

**Comment by adriancole** [comment]
>
> Adrian - That's a reasonable point, we can note in the comments that this
> is our language-independent encoding of TagContext that will also be used
> (minimally) by our C++ and Go implementations.
>
thx

> Unless there's a really big issue, changing the encoding is unlikely to
> happen at this point.
>
no issue on my side
